posthog/caching/fetch_from_cache.py

```diff
@@ -1,16 +1,23 @@
 from dataclasses import dataclass
-from datetime import datetime
-from typing import Any, Optional, Union
+from datetime import datetime, timedelta
+from math import ceil
+from typing import Any, Optional, Tuple, Union
 
+import pytz
 from django.utils.timezone import now
 from statshog.defaults.django import statsd
 
 from posthog.caching.calculate_results import calculate_cache_key, calculate_result_by_insight
 from posthog.caching.insight_cache import update_cached_state
+from posthog.caching.insight_caching_state import InsightCachingState
 from posthog.models import DashboardTile, Insight
 from posthog.models.dashboard import Dashboard
+from posthog.models.filters.utils import get_filter
 from posthog.utils import get_safe_cache
 
+# default minimum wait time for refreshing an insight
+DEFAULT_INSIGHT_REFRESH_FREQUENCY = timedelta(minutes=15)
```

Comment:
Problem: We have two concepts using that are similarly named:
- cache refreshes (running in celery)
- user-initiated refreshes.

It's not clear what applies to what. 

Suggestion: Let's make this module explicitly about insights api:
1. Add a docstring at the top saying these methods are used by insights/etc apis to fetch and decide whether to refresh results
2. Rename this file to `insights_api.py`?